frontend/packages/dev-console/src/components/pipelines/PipelineVisualizationTask.tsx

```diff
@@ -149,3 +156,22 @@ const TaskComponent: React.FC<TaskProps> = ({ task, status, name, isPipelineRun
     </li>
   );
 };
+
+interface TaskStatusProps {
+  steps: { name: string }[];
+  status: TaskStatus;
+}
+
+const TaskComponentTaskStatus: React.FC<TaskStatusProps> = ({ steps, status }) => {
+  if (steps.length === 0) return null;
+
+  const visualValues: StackedValue[] = steps.map((step) => {
+    return {
+      color: computeStepColor(step, status),
+      name: step.name,
+      size: 1,
+    };
+  });
+
+  return <HorizontalStackedBars values={visualValues} barGap={2} height={2} />;
```

Comment:
`barGap` and `height` being optional parameters, it's good to have either default values in the `horizontalStackedBars` component or making it a mandatory parameters. WDYT?

Reply:
They have defaults in the SCSS. Lack of applying them to the `style` prop allows the CSS to go with a default. `--bar-gap: 3px;` and `height: 100%` respectively.

Reply:
> If someone is not setting the height, the component will not be visually shown or won't get type errors.

No height is the box-model at work. If you don't have an explicit height, child items don't render with 100% height. Not unique to my design. If you wrap the bars in a fixed-height container, it'll take the full size.